modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/NameClashModelValidator.java

```diff
@@ -213,6 +256,49 @@ private void validateClash(String existingNamingModel, String newNamingModel, St
                                                          typeOfNewNamingModel));
       }
     }
+
+    private void validateSingularizedNameClash(ParameterizedModel model) {
+      List<ParameterModel> parameters = model.getParameterModels();
+      parameters.forEach(
+                         p -> p.getType().accept(new MetadataTypeVisitor() {
+
+                           @Override
+                           public void visitArrayType(ArrayType arrayType) {
+                             validateSingularizeNameClashWithOperationParameters();
+                           }
+
+                           @Override
+                           public void visitDictionary(DictionaryType dictionaryType) {
+                             validateSingularizeNameClashWithOperationParameters();
+                           }
+
+                           private void validateSingularizeNameClashWithOperationParameters() {
+                             String singularName = singularize(p.getName());
+                             boolean allowInline = p.getModelProperty(XmlHintsModelProperty.class)
+                                 .map(XmlHintsModelProperty::allowsInlineDefinition)
+                                 .orElse(true);
+
+                             if (!singularName.equals(p.getName()) && allowInline) {
+                               singularizedObjects.put(singularName, new DescribedParameter(p, model));
+                               Optional<ParameterModel> clashParam =
+                                   parameters.stream().filter(p -> p.getName().equals(singularName)).findAny();
+                               String describedReference = new DescribedReference<>(model).getDescription();
+                               Class<Object> type = getType(p.getType());
```

Comment:
there isn't necessarily a class behind this. This validator should also work for smart connectors